app/controllers/katello/api/v2/api_controller.rb

```diff
@@ -80,10 +80,15 @@ def scoped_search(query, default_sort_by, default_sort_order, options = {})
       total = query.count
       query = resource.search_for(*search_options).where("#{resource.table_name}.id" => query)
       sub_total = query.count
+
       sort_attr = params[:sort_by] || default_sort_by
-      sort_attr = "#{query.table_name}.#{sort_attr}" unless sort_attr.to_s.include?('.')
 
-      query = query.order("#{sort_attr} #{params[:sort_order] || default_sort_order}")
+      if sort_attr
+        sort_attr = "#{query.table_name}.#{sort_attr}" unless sort_attr.to_s.include?('.')
+        query = query.order("#{sort_attr} #{params[:sort_order] || default_sort_order}")
+      elsif options[:custom_sort]
```

Comment:
Maybe this should be checked for first since if you are providing this option, the intent would be for it to be processed , otherwise `sort_attr` should be processed?